Wikipedia/Code/WMFArticleContainerViewController+TOC.swift

```diff
@@ -19,19 +19,29 @@ extension WMFArticleContainerViewController : WMFTableOfContentsViewControllerDe
 
     public func tableOfContentsController(controller: WMFTableOfContentsViewController,
                                           didSelectItem item: TableOfContentsItem) {
-
+        var dismissVCCompletionHandler: (() -> Void)?
         if let section = item as? MWKSection {
             // HAX: webview has issues scrolling when browser view is out of bounds, disable animation if needed
             self.webViewController.scrollToSection(section, animated: self.webViewController.isWebContentVisible)
+            dismissVCCompletionHandler = {
+                // HAX: This is terrible, but iOS events not under our control would steal our focus if we didn't wait long enough here and due to problems in UIWebView, we cannot work around it either.
+                dispatchOnMainQueueAfterDelayInSeconds(1) {
+                    self.webViewController.accessibilityCursorToSection(section)
```

Comment:
Only using delay w/ accessibility w/ active sounds reasonable

On Tue, Feb 2, 2016 at 2:41 PM, hhanke <notifications@github.com> wrote:

> In Wikipedia/Code/WMFArticleContainerViewController+TOC.swift
> <https://github.com/wikimedia/wikipedia-ios/pull/439#discussion_r51623145>
> :
>
> >          if let section = item as? MWKSection {
> >              // HAX: webview has issues scrolling when browser view is out of bounds, disable animation if needed
> >              self.webViewController.scrollToSection(section, animated: self.webViewController.isWebContentVisible)
> > +            dismissVCCompletionHandler = {
> > +                // HAX: This is terrible, but iOS events not under our control would steal our focus if we didn't wait long enough here and due to problems in UIWebView, we cannot work around it either.
> > +                dispatchOnMainQueueAfterDelayInSeconds(1) {
> > +                    self.webViewController.accessibilityCursorToSection(section)
>
> @montehurd <https://github.com/montehurd> @bgerstle
> <https://github.com/bgerstle> I will try it, but I think unfortunately we
> can't go much under the 1 second as we would get the VO cursor stolen from
> us, as I described on the Phabricator :-/ However, I think we can work
> around the bug that @montehurd <https://github.com/montehurd> spotted
> (thank you!) by only invoking this when we know that an assistive
> technology is running. We can detect that. I think this is currently only
> relevant to VoiceOver, but I must give it a little more thinking. It will
> be a little less universal, but this solution is a hack anyway. That way it
> won't affect users not using assistive technologies at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/wikipedia-ios/pull/439/files#r51623145>.
>



-- 
EN Wikipedia user page: https://en.wikipedia.org/wiki/User:Brian.gerstle
IRC: bgerstle